CRM/Contribute/BAO/Contribution.php

```diff
@@ -5687,15 +5687,17 @@ public static function getSalesTaxFinancialAccounts() {
    * Create tax entry in civicrm_entity_financial_trxn table.
    *
    * @param array $entityParams
-   *
    * @param array $eftParams
    *
    */
   public static function createProportionalEntry($entityParams, $eftParams) {
-    $paid = $entityParams['line_item_amount'] * ($entityParams['trxn_total_amount'] / $entityParams['contribution_total_amount']);
-    // Record Entity Financial Trxn; CRM-20145
-    $eftParams['amount'] = CRM_Contribute_BAO_Contribution_Utils::formatAmount($paid);
-    civicrm_api3('EntityFinancialTrxn', 'create', $eftParams);
+    $contributionTotalAmount = (float) $entityParams['contribution_total_amount'];
+    if (!empty($contributionTotalAmount)) {
```

Comment:
@mattwire we need to create proportional entry in civicrm_entity_financial_trxn table even though the contribution total is 0.00 with civicrm_entity_financial_trxn.amount as 0.00. If there are no such entries then there would be problem while assigning transactions in financial batch or hopefully in quickbook reports as this transaction won't be listed because of missing entries.

Reply:
@pradnayak Ok. Happy to go with your PR if that's the case